src/Game.cxx

```diff
@@ -356,13 +371,16 @@ void Game::shutdown()
   UIManager::instance().flush();
 }
 
-template <typename MQType, typename Visitor> void Game::LoopMain(GameContext &context, Visitor visitor)
+template <typename MQType, typename Visitor>
+void Game::LoopMain(GameContext &context, Visitor visitor, typename MQType::Enumerable getEvents(GameContext &context))
```

Comment:
I don't think you need the callback parameter here. You could just create a policy or call `getEvents<MQType>()` directly instead

Reply:
I implemented the template function overload.